### PR TITLE
Add Python templates

### PR DESCRIPTION
**Pull request by justinvp**
Unfortunately [the CLI currently always attempts to run `npm install`, even for non-Node.js templates](https://github.com/pulumi/pulumi/issues/4115), so `pulumi policy new` will return the following error when a Python template is selected:

```
error: npm install failed; rerun manually to try again.: npm install reported success, but node_modules directory is missing
```

In this case, the error is benign and can be ignored.

We can hold off on merging this until we've had more time for the Python bits to bake, if we'd like. For the 3/19 Hackathon, folks can run:

```bash
pulumi policy new https://github.com/pulumi/templates-policy/tree/justin/python
```

### CODE REVIEW

File: azure-python/__main__.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    validate=storage_container_no_public_read_validator,
+)
+
+PolicyPack(
+    name="azure-python",
+    enforcement_level=EnforcementLevel.MANDATORY,
+    policies=[
+        storage_container_no_public_read,
+    ],
+)
```
Comment (lukehoban) [inline azure-python/__main__.py]: These are totally fine as is - but curious whether you think this or the the class-based style should be our default for templates/examples?

Comment (justinvp) [reply azure-python/__main__.py]: Good point. The approach I took here will be familiar to folks who are already familiar with writing policies in TypeScript/JavaScript.

Do you think the class-based style will feel more natural/idiomatic for Python developers? If so, we should go with that.

Here's a comparison of the two supported styles:

```python
def s3_no_public_read_validator(args: ResourceValidationArgs, report_violation: ReportViolation):
    if args.resource_type == "aws:s3/bucket:Bucket" and "acl" in args.props:
        acl = args.props["acl"]
        if acl == "public-read" or acl == "public-read-write":
            report_violation(
                "You cannot set public-read or public-read-write on an S3 bucket. " +
                "Read more about ACLs here: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html")

s3_no_public_read = ResourceValidationPolicy(
    name="s3-no-public-read",
    description="Prohibits setting the publicRead or publicReadWrite permission on AWS S3 buckets.",
    validate=s3_no_public_read_validator,
)

PolicyPack(
    name="aws-python",
    enforcement_level=EnforcementLevel.MANDATORY,
    policies=[
        s3_no_public_read,
    ],
)
```

vs

```python
class S3NoPublicReadPolicy(ResourceValidationPolicy):
    def __init__(self):
        super().__init__(
            name="s3-no-public-read",
            description="Prohibits setting the publicRead or publicReadWrite permission on AWS S3 buckets.")

    def validate(self, args: ResourceValidationArgs, report_violation: ReportViolation):
        if args.resource_type == "aws:s3/bucket:Bucket" and "acl" in args.props:
              acl = args.props["acl"]
              if acl == "public-read" or acl == "public-read-write":
                  report_violation("You cannot set public-read or public-read-write on an S3 bucket.")

PolicyPack(
      name="aws-policy-pack",
      enforcement_level=EnforcementLevel.MANDATORY,
      policies=[
          S3NoPublicReadPolicy(),
      ],
  )
```

Comment (ekrengel) [reply azure-python/__main__.py]: > more natural/idiomatic for Python developers?

Im not a Python dev.. but generally classes are used for objects that have methods on them, so I am included to use functions here.

Comment (lukehoban) [reply azure-python/__main__.py]: Sounds good - let's keep the function-based style for now then.  Definitely keeps consistency better, which will make explaining the APIs in documentation much easier.

**Pull request merged**